sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -378,11 +378,18 @@ object GpuCast extends Arm {
       // ansi cast from larger-than-long integral-like types, to long
       case (dt: DecimalType, LongType) if ansiMode =>
         // This is a work around for https://github.com/rapidsai/cudf/issues/9282
-        val min = BigDecimal(Long.MinValue)
+        val bigDecimalMin = BigDecimal(Long.MinValue)
             .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
-        val max = BigDecimal(Long.MaxValue)
+        val bigDecimalMax = BigDecimal(Long.MaxValue)
             .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
-        assertValuesInRange(input, Scalar.fromDecimal(min), Scalar.fromDecimal(max))
+        withResource(input.min()) { min =>
+          withResource(input.max()) { max =>
+            if (min.getBigDecimal().compareTo(bigDecimalMin) == -1 ||
+                max.getBigDecimal().compareTo(bigDecimalMax) == 1) {
+              throw new IllegalStateException(GpuCast.INVALID_INPUT_MESSAGE)
```

Comment:
this same thing throws in Spark on Cpu?

Reply:
Maybe we need to add an integration test to compare?

Reply:
> this same thing throws in Spark on Cpu?

CPU throws an ArithmeticException but I wanted to match the existing behavior. I have changed it to match the CPU. 

> Maybe we need to add an integration test to compare?

Just adding an integration test to for a handful of values or testing a wider range? I experimented with this and it will be a lot more involved test as there are many other types of exceptions that can be thrown in ANSI e.g. overflow

Reply:
I can imagine that this will be the case for other cast operations.
If we do not have integration tests to cover the behavior of CPU vs GPU, then we can create a new followup issue to improve the tests including the "Decimal to Long".

Reply:
+1 for adding a test to make sure that we are not regressing on the exception.